code/game/objects/barricades.dm

```diff
@@ -0,0 +1,379 @@
+
+	switch(tool_type)
+		if(QUALITY_WELDING)
+			if(health < maxHealth)
+				if(I.use_tool(user, src, WORKTIME_NORMAL, tool_type, FAILCHANCE_NORMAL,  required_stat = STAT_MEC))
+					user.visible_message(SPAN_NOTICE("\The [user] repairs \the [src]."), SPAN_NOTICE("You repair \the [src]."))
+					health = maxHealth
+					is_damaged = FALSE
+					update_icon()
+					return
+		if(QUALITY_BOLT_TURNING) // Deconstruction and hacking be harder when barrier is deployed
+			if(I.use_tool(user, src, (is_open ? WORKTIME_LONG : WORKTIME_NORMAL), tool_type, (is_open ? FAILCHANCE_CHALLENGING : FAILCHANCE_NORMAL),  required_stat = STAT_MEC))
+				for(var/material_name in matter)
+					var/material/material_datum = get_material_by_name(material_name)
+					material_datum.place_sheet(loc, amount = matter[material_name])
+				user.visible_message(SPAN_NOTICE("\The [user] dismantles \the [src]."), SPAN_NOTICE("You dismantle \the [src]."))
+				qdel(src)
+				return
+		if(QUALITY_PULSING)
+			if(I.use_tool(user, src, (is_open ? WORKTIME_LONG : WORKTIME_NORMAL), tool_type, (is_open ? FAILCHANCE_CHALLENGING : FAILCHANCE_NORMAL),  required_stat = STAT_MEC))
+				update_wifi_password()
+				return
+
+	user.do_attack_animation(src)
+	if(I.hitsound)
+		playsound(loc, I.hitsound, 50, 1, -1)
+	visible_message(SPAN_DANGER("[src] has been hit by [user] with [I]."))
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+	take_damage(I.force * I.structure_damage_factor)
+
+
+/obj/structure/barrier/four_way/receive_signal(datum/signal/signal, receive_method, receive_param)
+	ASSERT(signal)
+	if(signal.encryption == code)
+		toggle_barrier()
+
+
+/obj/structure/barrier/four_way/proc/toggle_barrier()
+	is_open = !is_open
+	density = !density
+	playsound(loc, 'sound/machines/Custom_bolts.ogg', 20)
+	update_icon()
+	flick(icon_state + "_animation", src)
+
+
+/obj/structure/barrier/hedgehog
+	name = "Serb hedgehog"
+	desc = "Static anti-mech barrier. Could be climbed over."
+	description_info = "Repaired by welding and dismantled with a tool capable of bolt turning."
+	icon_state = "hedgehog"
+	density = TRUE
+	climbable = TRUE
+	matter = list(MATERIAL_STEEL = 15)
+	health = 500
+	maxHealth = 500
+	resistance = RESISTANCE_TOUGH
+
+
+/obj/structure/barrier/barbed_wire
+	name = "barbed wire"
+	desc = "A mesh of metal strips with sharp edges, meant to prevent trespassing."
+	description_info = "Repaired with pliers and dismantled with a wire cutting tool."
+	icon_state = "barbed_wire"
+	block_vehicles = FALSE
+	matter = list(MATERIAL_STEEL = 10)
+	var/tresspass_damage = 15 // How much brute is dealt when someone tries to cross or attack the wire
+
+
+/obj/structure/barrier/barbed_wire/attackby(obj/item/I, mob/user)
+	var/list/usable_qualities = list(QUALITY_WIRE_CUTTING)
+	if(health < maxHealth)
+		usable_qualities.Add(QUALITY_CLAMPING)
+
+	var/tool_type = I.get_tool_type(user, usable_qualities)
+
+	if(tool_type == QUALITY_CLAMPING) // Pliers
```

Comment:
Are you sure clamping is a good quality for this? Clamping is supposed to be a medical-related quality, and pliers a shitty jack-of-all-trades.

Reply:
So it would seem? No strong feelings about this, just pick another quality, if this one is not to your liking.